discoverer/src/main/java/eu/seaclouds/platform/discoverer/core/Discoverer.java

```diff
@@ -17,250 +17,182 @@
 
 package eu.seaclouds.platform.discoverer.core;
 
-/* std imports */
-import java.io.*;
-import java.util.Calendar;
-import java.util.Date;
-import java.util.Iterator;
-import java.lang.Exception;
+import eu.seaclouds.platform.discoverer.api.*;
+import eu.seaclouds.platform.discoverer.Crawler.CrawlerManager;
+import io.dropwizard.Application;
+import io.dropwizard.setup.Environment;
 
-/* a4c */
-import alien4cloud.tosca.parser.ParsingException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
 
 
-public class Discoverer {
+public class Discoverer	extends Application<DiscovererConfiguration> {
 	/* singleton */
 	private static final Discoverer singleInstance = new Discoverer();
-	public static Discoverer instance() { return singleInstance; }
-	
+    private ArrayList<String> offeringNodeTemplates = new ArrayList<>();
+    private boolean refreshing = false;
+    private ArrayList<String> activeCrawlers;
+
+    public int totalCrawledOfferings;
+    public int crawledTimes;
+    public Date lastCrawl;
+
+    public static Discoverer instance() { return singleInstance; }
+
 	/* vars */
-	private UniqueFileGen ufg; // offerings ID generator
-	
-	
-	
+	public OfferingManager offeringManager;
+
+	private static IAPI[] APIList = {
```

Comment:
remove  this static API list. Initialize the Resources inside the `run` method as the resources may be configured using the `DiscovererConfiguration`
see [Registering a resource](http://www.dropwizard.io/0.9.1/docs/getting-started.html#registering-a-resource) for more details